tests/test_cache_lock_file_perf.py

```diff
@@ -0,0 +1,110 @@
+import multiprocessing
+import threading
+import time
+from msal_extensions import FilePersistence, CrossPlatLock
+
+
+def validate_thread_locking_file(num):
+    with open("C:/Users/abpati/Desktop/test/lockfilelog.txt") as handle:
+        data = handle.read()
+    time_list = []
+    for line in data.split("\n"):
+        start, end = line.split("-")
+        time_list.append([start, end])
+    if len(time_list) != num:
+        print("Threads count dont match ")
+        return
+    time_list.sort()
+    prev = None
+    for interval in time_list:
+        start, end = interval
+        if start > end:
+            print("Intervals start bigger than end")
+            return
+        if prev:
+            if start < prev:
+                print("overlapping")
+        prev = start
+    print("success")
+    return
+
+
+def validate_result_in_cache(num):
+    with open("C:/Users/abpati/Desktop/test/msal.cache") as handle:
+        data = handle.read()
+    prev_tag= None
+    prev_proc_id = None
+    count = 0
+    for line in data.split(("\n")):
+        tag, proc_id = line.split(" ")
+        if tag == '<':
+            if '<' == prev_tag:
+                print("Failed")
+                return
+        elif tag == '>':
+            count+=1
+            if '<' != prev_tag or (not prev_proc_id == proc_id):
+                print("Failed")
+                return
+        else:
+            print("Unexpected Token")
+        prev_proc_id = proc_id
+        prev_tag = tag
+    if ">" != prev_tag:
+        print("Failed")
+        return
+    if count != num:
+        print("Failed")
+        return
+    print("sucess")
+
+
+def do_something(tid):
+    lock_intervals_file_path = "C:/Users/abpati/Desktop/test/lockfilelog.txt"
+    lock_file_path = "C:/Users/abpati/Desktop/test/msal.cache.lockfile"
+    cache_accessor = FilePersistence("C:/Users/abpati/Desktop/test/msal.cache")
+    with CrossPlatLock(lock_file_path, lock_intervals_file_path):
+        thr = str(tid)
+        data = cache_accessor.load()
+        if data is None:
+            data = ""
+        data += "< " + thr + "\n"
+        time.sleep(0.1)
+        data += "> " + thr + "\n"
+        cache_accessor.save(data)
+
+
+def multiple_threads():
```

Comment:
@rayluo - it's interesting that both Java and Python ensure both cross-process and cross-thread concurrency. In .NET, we only ensure cross-process, because MSAL takes care of cross-thread. Just an observation that you may be locking too much...

Reply:
I also had to confirm with @SomkaPe during our daily stand-up yesterday to understand why Java implementation did this. His answer was basically "it simple to also implement multi-thread testing, so we just did it".

My take on this:

> "you may be locking too much..."

Originally I thought @bgavrilMS you mean our production implementation might lock too much. That would not be the case, because our real code path handles lock in exact same way, regardless of being used in multi-thread setup or multi-process setup.

And then I realize you actually mean "we may be *applying* locking too much *during our test*". It is true that, from MSAL EX's perspective, there would be no real need to test multi-thread scenario, because this MSAL EX is meant to be used in multi-process scenario. To that end, perhaps this PR would eventually only automate a multi-process test case with mild workload, which serves as a "known safe limit".

But then, given that part of the motivation of this test is to prove whether that hard limit exists, it doesn't harm to have a multi-thread capability (which is more efficient than multi-process), for us to manually run it and quickly hit that upper bound.